openstack/blockstorage/v3/volumetypes/requests.go

```diff
@@ -101,3 +101,38 @@ func List(client *gophercloud.ServiceClient, opts ListOptsBuilder) pagination.Pa
 		return VolumeTypePage{pagination.LinkedPageBase{PageResult: r}}
 	})
 }
+
+// UpdateOptsBuilder allows extensions to add additional parameters to the
+// Update request.
+type UpdateOptsBuilder interface {
+	ToVolumeTypeUpdateMap() (map[string]interface{}, error)
+}
+
+// UpdateOpts contain options for updating an existing Volume Type. This object is passed
+// to the volumetypes.Update function. For more information about the parameters, see
+// the Volume Type object.
+type UpdateOpts struct {
+	Name        string `json:"name,omitempty"`
+	Description string `json:"description,omitempty"`
+	IsPublic    bool   `json:"is_public"`
```

Comment:
This should be `*bool` and the json tag should be `json:"is_public,omitempty"`

`bool`s are tricky to work with. Let's say someone wants to update the name of a public volume type, so they do:

```go
updateOpts := volumetypes.UpdateOpts{
  Name: "new name",
}
```

But the request that would be sent to Cinder would be:

```json
{
  "volume_type": {
    "name": "new name",
    "is_public": false
  }
}
```

You could use `omitempty`, but that means no one could ever change a volume type from public to private because `false` would always be omitted.

Therefore, using `*bool` allows for three values: true, false, and nil. `omitempty` prevents an empty/null field from being sent to Cinder when the value is `nil`.

Search the Gophercloud repo for `iTrue` to see how passing values to bool-pointers is usually handled.